R/fit_model.R

```diff
@@ -0,0 +1,42 @@
+#' Fit the RtGam model with {mgcv}
+#'
+#' Use the pre-prepared model dataset and formula. Supply warnings as needed
+#' @noRd
+fit_model <- function(data, formula, backend) {
+  args <- args_constructor(data, formula, backend)
+  call <- call_constructor(backend)
+
+  do.call(
+    call,
+    args
+  )
+}
+
+args_constructor <- function(data, formula, backend) {
+  backend_agnostic_args <- list(
+    formula = formula,
+    data = data,
+    # Negative binomial family with overdispersion param estimated
+    family = "nb"
```

Comment:
yuor enforcing it to be NB? What if I want to look at a Poisson Gamma mixture (more likely just a random effect in the formula) or non-count data? We are just saying no?